### PR TITLE
Add in minimal retry metrics

### PR DESCRIPTION
**Pull request by revans2**
This depends on https://github.com/NVIDIA/spark-rapids-jni/pull/1031 and fixes https://github.com/NVIDIA/spark-rapids/issues/7880

I will file some follow on issues for the other metrics that were mice to have ones.

### CODE REVIEW

**Comment by revans2** [comment]
The dependency was just merged, but we have to wait for CI to get it published to where we need it before this can be marked as ready. Will probably do that tomorrow.

File: docs/tuning-guide.md [hunk 1]
```diff
@@ -345,14 +345,25 @@ How long does operator X take on the GPU vs the CPU?
...
+correspond to the data that was spilled by this task.
 
 | Name              | Description                                            |
 |-------------------|--------------------------------------------------------|
 | gpuSemaphoreWait  | The time the task spent waiting on the GPU semaphore.  |
 | gpuSpillBlockTime | The time that this task was blocked spilling data from the GPU. |
 | gpuSpillReadTime  | The time that this task was blocked reading data to the GPU that was spilled previously. |
+| gpuRetryCount | The number of times that a retry exception was thrown in an attempt to roll back processing to free memory. |
+| gpuSplitAndRetryCount | The number of times that a split and retry exception was thrown in an attempt to roll back processing to free memory, and split the input to make more room. |
+| gpuRetryBlockTime | The amount of time that this task was blocked ether hoping that other tasks will free up more memory or after a retry exception was thrown to wait until the task can go on. |
```
Comment (abellina) [inline docs/tuning-guide.md:366]:
```suggestion
| gpuRetryBlockTime | The amount of time that this task was blocked either hoping that other tasks will free up more memory or after a retry exception was thrown to wait until the task can go on. |
```

**Review by abellina — COMMENTED** [review]
one minor typo but LGTM

**Review by jbrennan333 — APPROVED** [review]
LGTM

**Comment by revans2** [comment]
build

**Review by abellina — APPROVED** [review]

**Comment by revans2** [comment]
build

**Comment by revans2** [comment]
build

**Comment by revans2** [comment]
build

**Pull request merged**